src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -142,54 +166,81 @@ const useFetchChannelHook = () => {
 
       (members || []).map((member) => {
         const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
-        queue.addJob({
-          label: `saveUserMember-${member?.user?.id}`,
-          task: () => {
-            return new Promise((resolve) => {
-              UserSchema.isUserExists(localDb, member?.user?.id, channel?.id).then(
-                (isUserExists) => {
-                  if (!member?.user?.username) return resolve(true);
-                  if (!isUserExists) {
-                    userMember.save(localDb).then(() => {
+
+        if (Platform.OS === 'android') {
+          UserSchema.isUserExists(localDb, member?.user?.id, channel?.id).then((isUserExists) => {
+            if (!member?.user?.username) return;
+            if (!isUserExists) {
+              userMember.save(localDb);
+            }
+          });
+        } else {
+          queue.addJob({
+            label: `saveUserMember-${member?.user?.id}`,
+            task: () => {
+              return new Promise((resolve) => {
+                UserSchema.isUserExists(localDb, member?.user?.id, channel?.id).then(
+                  (isUserExists) => {
+                    if (!member?.user?.username) return resolve(true);
+                    if (!isUserExists) {
+                      userMember.save(localDb).then(() => {
+                        return resolve(true);
+                      });
+                    } else {
                       return resolve(true);
-                    });
-                  } else {
-                    return resolve(true);
+                    }
                   }
-                }
-              );
-            });
-          }
-        });
+                );
+              });
+            }
+          });
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block introduces a platform-specific conditional check for saving user members. On Android, it directly checks if the user exists and saves the user member if not. On other platforms, it adds this task to a queue for later processing. This seems like an optimization for Android, but it's important to ensure that this doesn't introduce any race conditions or inconsistencies in the application state.

```diff
+         if (Platform.OS === 'android') {
+           UserSchema.isUserExists(localDb, member?.user?.id, channel?.id).then((isUserExists) => {
+             if (!member?.user?.username) return;
+             if (!isUserExists) {
+               userMember.save(localDb);
+             }
+           });
+         } else {
+           queue.addJob({
+             label: `saveUserMember-${member?.user?.id}`,
+             task: () => {
+               return new Promise((resolve) => {
+                 UserSchema.isUserExists(localDb, member?.user?.id, channel?.id).then(
+                   (isUserExists) => {
+                     if (!member?.user?.username) return resolve(true);
+                     if (!isUserExists) {
+                       userMember.save(localDb).then(() => {
+                         return resolve(true);
+                       });
+                     } else {
+                        return resolve(true);
+                     }
+                    }
+                 );
+               });
+             }
+           });
+         }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->